docs/_docs/reference/contextual/context-bounds.md

```diff
@@ -4,50 +4,208 @@ title: "Context Bounds"
+
+ 2. By rule 2,
+    ```scala
+    def f[T: {C1, C2}, U: C3](x: T)(using y: U, z: V): R
+    ```
+    expands to
+    ```scala
+    def f[T, U](x: T)(using _: C1[T], _: C2[T], _: C3[U], y: U, z: V): R
+    ```
+    The same expansion occurs if `y` and `z` are Scala 2 style `implicit` parameters.
+ 3. Assume the following trait definition:
+    ```scala
+    trait Parser[P]:
+      type Input
+      type Result
+    ```
+    Here is a method `run` that runs a parser on an input of the required type:
+    ```scala
+    def run[P : Parser as p](in: p.Input): p.Result
+    ```
+    By rule 1, this method definition is expanded to:
+    ```scala
+    def run[P](using p: Parser[P]](in: p.Input): p.Result
+    ```
+    Note that the `using` clause is placed in front of the explicit parameter clause `(in: p.Result)` so that
+    the type `p.Result` can legally refer to the context parameter `p`.
+
+### Migration
+
+To ease migration, context bounds map in Scala 3.0 - 3.5 to old-style implicit parameters
+for which arguments can be passed either with a `(using ...)` clause or with a normal application. From Scala 3.6 on, they will map to context parameters instead, as is described above.
+
+If the source version is `3.6-migration`, any pairing of an evidence
 context parameter stemming from a context bound with a normal argument will give a migration
 warning. The warning indicates that a `(using ...)` clause is needed instead. The rewrite can be
 done automatically under `-rewrite`.
 
+## Context Bounds for Polymorphic Functions
+
+From Scala 3.6 on, context bounds can also be used in polymorphic function types and polymorphic function literals:
+
+```scala
+type Comparer = [X: Ord] => (x: X, y: X) => Boolean
+val less: Comparer = [X: Ord as ord] => (x: X, y: X) =>
+  ord.compare(x, y) < 0
+```
+
+The expansion of such context bounds is analogous to the expansion in method types, except that instead of adding a using clause in a method, we insert a [context function type](./context-functions.md).
+
+For instance, the `type` and `val` definitions above would expand to
+```scala
+type Comparer = [X] => (x: X, y: X) => Ord[X] ?=> Boolean
+val less: Comparer = [X] => (x: X, y: X) => (ord: Ord[X]) ?=>
+  ord.compare(x, y) < 0
+```
+
+The expansion of using clauses does look inside alias types. For instance,
+here is a variation of the previous example that uses a parameterized type alias:
+```scala
+type Cmp[X] = (x: X, y: X) => Boolean
+type Comparer2 = [X: Ord] => Cmp[X]
+```
+The expansion of the right hand side of `Comparer2` expands the `Cmp[X]` alias
+and then inserts the context function at the same place as what's done for `Comparer`:
+```scala
+  [X] => (x: X, y: X) => Ord[X] ?=> Boolean
+```
+
+### Context Bounds for Type Members
+
+From Scala 3.6 on, context bounds can not only used for type parameters but also for abstract type members.
+
+**Example**:
+
+```scala
+  class Collection:
+    type Element: Ord
+```
```

Comment:
This example might be incorrect aswell. I'm not sure it it's a bug or expected behaviour, but it fails to compile with 
```scala
[error] No given instance of type Ord[Sorted.this.Element] was found for inferring the implementation of the deferred given instance given_Ord_Element in class Sorted
```

It does however compile when replacing `class Collection` with `trait Collection`

Reply:
Oh yes, that's a bug. It should be trait.